libs/api_common/registry_helper.cpp

```diff
@@ -0,0 +1,239 @@
+#include <codecvt>
+#include <map>
+#include <mutex>
+#include <stdexcept>
+#include "api_common.hpp"
+#include "device_helper.hpp"
+#include "ebpf_api.h"
+#include "ebpf_platform.h"
+#include "ebpf_program_types.h"
+#include "ebpf_protocol.h"
+#include "ebpf_result.h"
+#include "platform.h"
+#include "platform.hpp"
+#include "um_registry_helper.h"
+
+#define GUID_STRING_LENGTH 38 // not inlcuding the null terminator.
+
+static std::wstring
+_get_wstring_from_string(std::string text)
+{
+    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+    std::wstring wide = converter.from_bytes(text);
+
+    return wide;
+}
+
+void
+close_registry_key(_In_ ebpf_registry_key_t* key)
+{
+    if (key->key != nullptr) {
+        RegCloseKey(key->key);
+        key->key = nullptr;
+    }
+}
+
+uint32_t
+write_registry_value_binary(
+    _In_ const ebpf_registry_key_t* key,
+    _In_ const wchar_t* value_name,
+    _In_reads_(value_size) uint8_t* value,
+    _In_ size_t value_size)
+{
+    ebpf_assert(value_name);
+    ebpf_assert(value);
+
+    return RegSetValueEx(key->key, value_name, 0, REG_BINARY, value, (DWORD)value_size);
+}
+
+uint32_t
+write_registry_value_wide_string(
+    _In_ const ebpf_registry_key_t* key, _In_ const wchar_t* value_name, _In_z_ const wchar_t* value)
+{
+    ebpf_assert(value_name);
+    ebpf_assert(value);
+
+    auto length = (wcslen(value) + 1) * sizeof(wchar_t);
+    return RegSetValueEx(key->key, value_name, 0, REG_SZ, (uint8_t*)value, (DWORD)length);
+}
+
+uint32_t
+write_registry_value_ansi_string(
+    _In_ const ebpf_registry_key_t* key, _In_ const wchar_t* value_name, _In_z_ const char* value)
+{
+    uint32_t result;
+    try {
+        auto wide_string = _get_wstring_from_string(value);
+        result = write_registry_value_wide_string(key, value_name, wide_string.c_str());
+    } catch (...) {
+        result = ERROR_NOT_ENOUGH_MEMORY;
+    }
+
+    return result;
+}
+
+uint32_t
+write_registry_value_dword(_In_ const ebpf_registry_key_t* key, _In_z_ const wchar_t* value_name, uint32_t value)
+{
+    ebpf_assert(key->key);
+    return RegSetValueEx(key->key, value_name, 0, REG_DWORD, (PBYTE)&value, sizeof(value));
+}
+
+uint32_t
+create_registry_key(
+    _In_opt_ const ebpf_registry_key_t* root_key,
+    _In_ const wchar_t* sub_key,
+    uint32_t flags,
+    _Out_ ebpf_registry_key_t* key)
+{
+    key->key = nullptr;
+    if (root_key == nullptr) {
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    return RegCreateKeyEx(root_key->key, sub_key, 0, nullptr, 0, flags, nullptr, &key->key, nullptr);
+}
+
+uint32_t
+open_registry_key(
+    _In_ const ebpf_registry_key_t* root_key,
+    _In_opt_z_ const wchar_t* sub_key,
+    uint32_t flags,
+    _Out_ ebpf_registry_key_t* key)
+{
+    ebpf_assert(root_key != nullptr);
+
+    return RegOpenKeyEx(root_key->key, sub_key, 0, flags, &key->key);
+}
+
+uint32_t
+delete_registry_key(_In_ const ebpf_registry_key_t* root_key, _In_z_ const wchar_t* sub_key)
```

Comment:
what is the value of these one line wrappers?

Reply:
Added these wrappers for consistency with other registry APIs. Also this API will be used in future ebpf store APIs to delete the entries from the store (in another PR), so these "read / delete" APIs will need wrappers so that calling code can execute both in user mode and kernel mode.